File: Praktikum 6. Array/Soal 1/PRAK601-2310817310001-AvantioFierzaPatria.c
#include <stdio.h>

void buatMatriks()
{
    int baris, kolom;

    printf("Masukkan jumlah baris dan kolom (pisahkan dengan spasi): ");
    scanf("%d %d", &baris, &kolom);

    int matriks[baris][kolom];

    printf("Masukkan angka-angka dalam matriks:\n");
    for (int i = 0; i < baris; i++)
    {
        for (int j = 0; j < kolom; j++)
        {
            scanf("%d", &matriks[i][j]);
        }
    }

    printf("Matriks yang dibuat:\n");
    for (int i = 0; i < baris; i++)
    {
        for (int j = 0; j < kolom; j++)
        {
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

int main()
{
    buatMatriks();

    return 0;
}

File: Praktikum 6. Array/Soal 1/PRAK601-2310817310001-AvantioFierzaPatria.py
def buat_matriks():

    baris, kolom = map(int, input(
        "Masukkan jumlah baris dan kolom (pisahkan dengan spasi): ").split())
    angka_matriks = list(map(int, input(
        "Masukkan angka-angka dalam matriks (pisahkan dengan spasi): ").split()))

    if len(angka_matriks) != baris * kolom:
        print("Jumlah angka tidak sesuai dengan baris dan kolom yang ditetapkan.")
        return None

    matriks = [angka_matriks[i:i + kolom]
               for i in range(0, len(angka_matriks), kolom)]

    print("Matriks yang dibuat:")
    for baris_matriks in matriks:
        print(baris_matriks)

    return matriks


matriks_midoriya = buat_matriks()
